### ISSUE TITLE
Default code renderer should renderer attributes

### ISSUE THREAD

**Issue by jugglingcats** [markdown-it/markdown-it#261]
I was trying to add an attribute to `code_inline` using a decorator style approach. My code looked something like this:

```javascript
            md.core.ruler.push('ev-doc-code-decorator', function (state) {
                for (var i = 0; i < state.tokens.length; i++) {
                    var token = state.tokens[i];
                    if (token.type === 'inline' && token.children ) {
                        for (var j = 0; j < token.children.length; j++) {
                            var child = token.children[j];
                            if (child.type === 'code_inline') {
                              token.attrSet('ng-non-bindable', '');
                            }
                    }
                }
            });
```

Unfortunately this doesn't work, because the default code_inline renderer doesn't take account of any attributes:

```javascript
    default_rules.code_inline = function (tokens, idx /*, options, env */) {
      return '<code>' + escapeHtml(tokens[idx].content) + '</code>';
    };
```

I managed to workaround this using the following code:

```javascript
            var orig=md.renderer.rules.code_inline;
            md.renderer.rules.code_inline = function(tokens, idx) {
                return '<span ng-non-bindable>' + orig(tokens, idx) + '</span>';
            };
```

It works, but am sure this is not the best approach...

Thanks

**Comment by puzrin** [comment]
Please check master. If it's ok, i will publish

**Comment by jugglingcats** [comment]
That's fast! Change looks good to me.

Can you advise whether there is a more efficient way to decorate inline tokens, ie. better than my first attempt above. Because my workaround, while being a complete hack, is a lot less code and arguably clearer to read.

Thanks

**Comment by puzrin** [comment]
I can't tell what is "better" for you. Override renderer function is normal, it's designed for that.

**Comment by jugglingcats** [comment]
Something like this would be nice:

```
  md.core.inline_ruler.push('ev-doc-code-decorator', function (token) {
    if (token.type === 'code_inline') {
      token.attrSet('ng-non-bindable', '');
    }
  }
```
(or similar)

I guess it is quite common to want to process the inline token pipeline without being concerned with rendering of wanting to override/wrap the default renderers.

**Comment by puzrin** [comment]
https://github.com/markdown-it/markdown-it-for-inline ?

**Comment by puzrin** [comment]
Released 6.1.1

**Comment by jugglingcats** [comment]
Thanks for link - looks useful - not sure why I didn't find it...
